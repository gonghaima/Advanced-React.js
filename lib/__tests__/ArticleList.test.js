import React from 'react';
import ArticleList from '../components/ArticleList';
import data from '../testData';
// import ReactTestRenderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

const testProps = {
  articles: data.articles,
  actions: {
    lookupAuthorById: jest.fn(() => data.authors[0] )
  }
};

describe('Article List Component', () => {
  it('should render correctly', () => {
    configure({ adapter: new Adapter() });
    const wrapper = shallow(
      <ArticleList {...testProps} />
    );
    // console.log('wrapper:',wrapper);
    // console.log('wrapper.e', wrapper.exists);
    // console.log('getEle', wrapper.getElement());
    // console.log('getEle', wrapper.getElement().props.children.length);
    expect(wrapper.getElement().props.children.length).toBe(5);
    
    expect(wrapper).toMatchSnapshot();
    // expect(domTree.children.length).toBeGreaterThan(0);
  });
});