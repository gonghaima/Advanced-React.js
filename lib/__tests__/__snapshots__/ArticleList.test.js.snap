// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article List Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList
    actions={
      Object {
        "lookupAuthorById": [MockFunction],
      }
    }
    articles={
      Array [
        Object {
          "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
          "body": "Building a better developer experience has been one of the things that React deeply cares about, and a crucial part of it is to detect anti-patterns/potential errors early and provide helpful error messages when things (may) go wrong. However, most of these only exist in development mode; in production, we avoid having extra expensive assertions and sending down full error messages in order to reduce the number of bytes sent over the wire.",
          "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
          "id": "95c12a8f6c88953ca8f8a39da25546e6",
          "title": "Introducing React's Error Code System",
        },
        Object {
          "authorId": "78ae672985c41fae0ecde0133f41bbfa",
          "body": "“How do I share the code between several components?” is one of the first questions that people ask when they learn React. Our answer has always been to use component composition for code reuse. You can define a component and use it in several other components. 
It is not always obvious how a certain pattern can be solved with composition. React is influenced by functional programming but it came into the field that was dominated by object-oriented libraries. It was hard for engineers both inside and outside of Facebook to give up on the patterns they were used to.",
          "date": "Wed Jul 13 2016 00:00:00 GMT+0000 (UTC)",
          "id": "cc7781c085cf37aabf120098085ff60c",
          "title": "Mixins Considered Harmful",
        },
        Object {
          "authorId": "78ae672985c41fae0ecde0133f41bbfa",
          "body": "Create React App is a new officially supported way to create single-page React applications. It offers a modern build setup with no configuration. 

Getting Starte 
Installation 
First, install the global package:",
          "date": "Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)",
          "id": "0a9afe5bb4ecbf4f7f1c77611e9bf1f9",
          "title": "Create Apps with No Configuration",
        },
        Object {
          "authorId": "335fb02ec8f76c8515821ac9f266d276",
          "body": "This month marks a year since we released Relay and we'd like to share an update on the project and what's next. 
A Year In Review 
A year after launch, we're incredibly excited to see an active community forming around Relay and that companies such as Twitter are using Relay in production:",
          "date": "Fri Aug 05 2016 00:00:00 GMT+0000 (UTC)",
          "id": "9b72140f27e62670dd7bdd1a9f61b48e",
          "title": "Relay: State of the State",
        },
        Object {
          "authorId": "d85577ea34ae50f2dac5347b5219aa23",
          "body": "It's been exactly one year since the last breaking change to React. Our next major release, React 16, will include some exciting improvements, including a complete rewrite of React's internals. We take stability seriously, and are committed to bringing those improvements to all of our users with minimal effort.
 To that end, today we're releasing React 15.5.0.",
          "date": "Fri Apr 07 2017 00:00:00 GMT+0000 (UTC)",
          "id": "f4ab6de6e61c86f03f6fef46f7c407f1",
          "title": "React v15.5.0",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Article
          article={
            Object {
              "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
              "body": "Building a better developer experience has been one of the things that React deeply cares about, and a crucial part of it is to detect anti-patterns/potential errors early and provide helpful error messages when things (may) go wrong. However, most of these only exist in development mode; in production, we avoid having extra expensive assertions and sending down full error messages in order to reduce the number of bytes sent over the wire.",
              "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
              "id": "95c12a8f6c88953ca8f8a39da25546e6",
              "title": "Introducing React's Error Code System",
            }
          }
        />,
        <Article
          article={
            Object {
              "authorId": "78ae672985c41fae0ecde0133f41bbfa",
              "body": "“How do I share the code between several components?” is one of the first questions that people ask when they learn React. Our answer has always been to use component composition for code reuse. You can define a component and use it in several other components. 
It is not always obvious how a certain pattern can be solved with composition. React is influenced by functional programming but it came into the field that was dominated by object-oriented libraries. It was hard for engineers both inside and outside of Facebook to give up on the patterns they were used to.",
              "date": "Wed Jul 13 2016 00:00:00 GMT+0000 (UTC)",
              "id": "cc7781c085cf37aabf120098085ff60c",
              "title": "Mixins Considered Harmful",
            }
          }
        />,
        <Article
          article={
            Object {
              "authorId": "78ae672985c41fae0ecde0133f41bbfa",
              "body": "Create React App is a new officially supported way to create single-page React applications. It offers a modern build setup with no configuration. 

Getting Starte 
Installation 
First, install the global package:",
              "date": "Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)",
              "id": "0a9afe5bb4ecbf4f7f1c77611e9bf1f9",
              "title": "Create Apps with No Configuration",
            }
          }
        />,
        <Article
          article={
            Object {
              "authorId": "335fb02ec8f76c8515821ac9f266d276",
              "body": "This month marks a year since we released Relay and we'd like to share an update on the project and what's next. 
A Year In Review 
A year after launch, we're incredibly excited to see an active community forming around Relay and that companies such as Twitter are using Relay in production:",
              "date": "Fri Aug 05 2016 00:00:00 GMT+0000 (UTC)",
              "id": "9b72140f27e62670dd7bdd1a9f61b48e",
              "title": "Relay: State of the State",
            }
          }
        />,
        <Article
          article={
            Object {
              "authorId": "d85577ea34ae50f2dac5347b5219aa23",
              "body": "It's been exactly one year since the last breaking change to React. Our next major release, React 16, will include some exciting improvements, including a complete rewrite of React's internals. We take stability seriously, and are committed to bringing those improvements to all of our users with minimal effort.
 To that end, today we're releasing React 15.5.0.",
              "date": "Fri Apr 07 2017 00:00:00 GMT+0000 (UTC)",
              "id": "f4ab6de6e61c86f03f6fef46f7c407f1",
              "title": "React v15.5.0",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "95c12a8f6c88953ca8f8a39da25546e6",
        "nodeType": "function",
        "props": Object {
          "article": Object {
            "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
            "body": "Building a better developer experience has been one of the things that React deeply cares about, and a crucial part of it is to detect anti-patterns/potential errors early and provide helpful error messages when things (may) go wrong. However, most of these only exist in development mode; in production, we avoid having extra expensive assertions and sending down full error messages in order to reduce the number of bytes sent over the wire.",
            "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
            "id": "95c12a8f6c88953ca8f8a39da25546e6",
            "title": "Introducing React's Error Code System",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "cc7781c085cf37aabf120098085ff60c",
        "nodeType": "function",
        "props": Object {
          "article": Object {
            "authorId": "78ae672985c41fae0ecde0133f41bbfa",
            "body": "“How do I share the code between several components?” is one of the first questions that people ask when they learn React. Our answer has always been to use component composition for code reuse. You can define a component and use it in several other components. 
It is not always obvious how a certain pattern can be solved with composition. React is influenced by functional programming but it came into the field that was dominated by object-oriented libraries. It was hard for engineers both inside and outside of Facebook to give up on the patterns they were used to.",
            "date": "Wed Jul 13 2016 00:00:00 GMT+0000 (UTC)",
            "id": "cc7781c085cf37aabf120098085ff60c",
            "title": "Mixins Considered Harmful",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0a9afe5bb4ecbf4f7f1c77611e9bf1f9",
        "nodeType": "function",
        "props": Object {
          "article": Object {
            "authorId": "78ae672985c41fae0ecde0133f41bbfa",
            "body": "Create React App is a new officially supported way to create single-page React applications. It offers a modern build setup with no configuration. 

Getting Starte 
Installation 
First, install the global package:",
            "date": "Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)",
            "id": "0a9afe5bb4ecbf4f7f1c77611e9bf1f9",
            "title": "Create Apps with No Configuration",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "9b72140f27e62670dd7bdd1a9f61b48e",
        "nodeType": "function",
        "props": Object {
          "article": Object {
            "authorId": "335fb02ec8f76c8515821ac9f266d276",
            "body": "This month marks a year since we released Relay and we'd like to share an update on the project and what's next. 
A Year In Review 
A year after launch, we're incredibly excited to see an active community forming around Relay and that companies such as Twitter are using Relay in production:",
            "date": "Fri Aug 05 2016 00:00:00 GMT+0000 (UTC)",
            "id": "9b72140f27e62670dd7bdd1a9f61b48e",
            "title": "Relay: State of the State",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "f4ab6de6e61c86f03f6fef46f7c407f1",
        "nodeType": "function",
        "props": Object {
          "article": Object {
            "authorId": "d85577ea34ae50f2dac5347b5219aa23",
            "body": "It's been exactly one year since the last breaking change to React. Our next major release, React 16, will include some exciting improvements, including a complete rewrite of React's internals. We take stability seriously, and are committed to bringing those improvements to all of our users with minimal effort.
 To that end, today we're releasing React 15.5.0.",
            "date": "Fri Apr 07 2017 00:00:00 GMT+0000 (UTC)",
            "id": "f4ab6de6e61c86f03f6fef46f7c407f1",
            "title": "React v15.5.0",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Article
            article={
              Object {
                "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
                "body": "Building a better developer experience has been one of the things that React deeply cares about, and a crucial part of it is to detect anti-patterns/potential errors early and provide helpful error messages when things (may) go wrong. However, most of these only exist in development mode; in production, we avoid having extra expensive assertions and sending down full error messages in order to reduce the number of bytes sent over the wire.",
                "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
                "id": "95c12a8f6c88953ca8f8a39da25546e6",
                "title": "Introducing React's Error Code System",
              }
            }
          />,
          <Article
            article={
              Object {
                "authorId": "78ae672985c41fae0ecde0133f41bbfa",
                "body": "“How do I share the code between several components?” is one of the first questions that people ask when they learn React. Our answer has always been to use component composition for code reuse. You can define a component and use it in several other components. 
It is not always obvious how a certain pattern can be solved with composition. React is influenced by functional programming but it came into the field that was dominated by object-oriented libraries. It was hard for engineers both inside and outside of Facebook to give up on the patterns they were used to.",
                "date": "Wed Jul 13 2016 00:00:00 GMT+0000 (UTC)",
                "id": "cc7781c085cf37aabf120098085ff60c",
                "title": "Mixins Considered Harmful",
              }
            }
          />,
          <Article
            article={
              Object {
                "authorId": "78ae672985c41fae0ecde0133f41bbfa",
                "body": "Create React App is a new officially supported way to create single-page React applications. It offers a modern build setup with no configuration. 

Getting Starte 
Installation 
First, install the global package:",
                "date": "Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)",
                "id": "0a9afe5bb4ecbf4f7f1c77611e9bf1f9",
                "title": "Create Apps with No Configuration",
              }
            }
          />,
          <Article
            article={
              Object {
                "authorId": "335fb02ec8f76c8515821ac9f266d276",
                "body": "This month marks a year since we released Relay and we'd like to share an update on the project and what's next. 
A Year In Review 
A year after launch, we're incredibly excited to see an active community forming around Relay and that companies such as Twitter are using Relay in production:",
                "date": "Fri Aug 05 2016 00:00:00 GMT+0000 (UTC)",
                "id": "9b72140f27e62670dd7bdd1a9f61b48e",
                "title": "Relay: State of the State",
              }
            }
          />,
          <Article
            article={
              Object {
                "authorId": "d85577ea34ae50f2dac5347b5219aa23",
                "body": "It's been exactly one year since the last breaking change to React. Our next major release, React 16, will include some exciting improvements, including a complete rewrite of React's internals. We take stability seriously, and are committed to bringing those improvements to all of our users with minimal effort.
 To that end, today we're releasing React 15.5.0.",
                "date": "Fri Apr 07 2017 00:00:00 GMT+0000 (UTC)",
                "id": "f4ab6de6e61c86f03f6fef46f7c407f1",
                "title": "React v15.5.0",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "95c12a8f6c88953ca8f8a39da25546e6",
          "nodeType": "function",
          "props": Object {
            "article": Object {
              "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
              "body": "Building a better developer experience has been one of the things that React deeply cares about, and a crucial part of it is to detect anti-patterns/potential errors early and provide helpful error messages when things (may) go wrong. However, most of these only exist in development mode; in production, we avoid having extra expensive assertions and sending down full error messages in order to reduce the number of bytes sent over the wire.",
              "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
              "id": "95c12a8f6c88953ca8f8a39da25546e6",
              "title": "Introducing React's Error Code System",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "cc7781c085cf37aabf120098085ff60c",
          "nodeType": "function",
          "props": Object {
            "article": Object {
              "authorId": "78ae672985c41fae0ecde0133f41bbfa",
              "body": "“How do I share the code between several components?” is one of the first questions that people ask when they learn React. Our answer has always been to use component composition for code reuse. You can define a component and use it in several other components. 
It is not always obvious how a certain pattern can be solved with composition. React is influenced by functional programming but it came into the field that was dominated by object-oriented libraries. It was hard for engineers both inside and outside of Facebook to give up on the patterns they were used to.",
              "date": "Wed Jul 13 2016 00:00:00 GMT+0000 (UTC)",
              "id": "cc7781c085cf37aabf120098085ff60c",
              "title": "Mixins Considered Harmful",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0a9afe5bb4ecbf4f7f1c77611e9bf1f9",
          "nodeType": "function",
          "props": Object {
            "article": Object {
              "authorId": "78ae672985c41fae0ecde0133f41bbfa",
              "body": "Create React App is a new officially supported way to create single-page React applications. It offers a modern build setup with no configuration. 

Getting Starte 
Installation 
First, install the global package:",
              "date": "Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)",
              "id": "0a9afe5bb4ecbf4f7f1c77611e9bf1f9",
              "title": "Create Apps with No Configuration",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "9b72140f27e62670dd7bdd1a9f61b48e",
          "nodeType": "function",
          "props": Object {
            "article": Object {
              "authorId": "335fb02ec8f76c8515821ac9f266d276",
              "body": "This month marks a year since we released Relay and we'd like to share an update on the project and what's next. 
A Year In Review 
A year after launch, we're incredibly excited to see an active community forming around Relay and that companies such as Twitter are using Relay in production:",
              "date": "Fri Aug 05 2016 00:00:00 GMT+0000 (UTC)",
              "id": "9b72140f27e62670dd7bdd1a9f61b48e",
              "title": "Relay: State of the State",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "f4ab6de6e61c86f03f6fef46f7c407f1",
          "nodeType": "function",
          "props": Object {
            "article": Object {
              "authorId": "d85577ea34ae50f2dac5347b5219aa23",
              "body": "It's been exactly one year since the last breaking change to React. Our next major release, React 16, will include some exciting improvements, including a complete rewrite of React's internals. We take stability seriously, and are committed to bringing those improvements to all of our users with minimal effort.
 To that end, today we're releasing React 15.5.0.",
              "date": "Fri Apr 07 2017 00:00:00 GMT+0000 (UTC)",
              "id": "f4ab6de6e61c86f03f6fef46f7c407f1",
              "title": "React v15.5.0",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
